src/qos_core/src/reaper.rs

```diff
@@ -164,15 +164,17 @@ impl Reaper {
 					.spawn()
 					.expect("Failed to spawn")
 					.wait()
+					.await
 					.expect("Pivot executable never started...");
 
 				println!("Pivot exited with status: {status}");
 
 				// pause to ensure OS has enough time to clean up resources
 				// before restarting
-				std::thread::sleep(std::time::Duration::from_secs(
+				tokio::time::sleep(std::time::Duration::from_secs(
 					REAPER_RESTART_DELAY_IN_SECONDS,
```

Comment:
I know this isn't directly related to this PR, but I think this should be something between 1 to 10 milliseconds to minimize service interruption if the app needs a restart. The number is kind of arbitrary but I just want this to be as quick as we can afford